File: ruby/bob/bob.rb
class Bob

  def hey(remark)
    if remark.strip.empty?
      'Fine. Be that way!'
    elsif remark == remark.upcase && has_letters?(remark)
      'Whoa, chill out!'
    elsif remark.end_with?('?')
      'Sure.'
    else
      'Whatever.'
    end
  end

  private
  def has_letters?(remark)
    !remark.gsub(/[^a-z]/i, '').empty?
  end

end

File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {
  if(input.trim().length === 0){
    return 'Fine. Be that way!';
  }
  if(input.toUpperCase() === input && /[A-Z]/.test(input)){
    return 'Whoa, chill out!';
  }
  if(input[input.length -1] === '?'){
    return 'Sure.';
  }
  return 'Whatever.';

};

module.exports = Bob;
